libsplinter/src/admin/store/diesel/operations/remove_circuit.rs

```diff
@@ -0,0 +1,46 @@
+// Copyright 2018-2020 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Provides the "remove circuit" operation for the `DieselAdminServiceStore`.
+
+use diesel::{dsl::delete, prelude::*};
+
+use crate::admin::store::{diesel::schema::circuit, error::AdminServiceStoreError};
+
+use super::{fetch_circuit::AdminServiceStoreFetchCircuitOperation, AdminServiceStoreOperations};
+
+pub(in crate::admin::store::diesel) trait AdminServiceStoreRemoveCircuitOperation {
+    fn remove_circuit(&self, circuit_id: &str) -> Result<(), AdminServiceStoreError>;
+}
+
+impl<'a, C> AdminServiceStoreRemoveCircuitOperation for AdminServiceStoreOperations<'a, C>
+where
+    C: diesel::Connection,
+    String: diesel::deserialize::FromSql<diesel::sql_types::Text, C::Backend>,
+    i64: diesel::deserialize::FromSql<diesel::sql_types::BigInt, C::Backend>,
+{
+    fn remove_circuit(&self, circuit_id: &str) -> Result<(), AdminServiceStoreError> {
+        // Verify the circuit attempting to be removed exists.
+        self.fetch_circuit(&circuit_id).and_then(|_| {
```

Comment:
Because the fetch performs its own queries, this whole thing should be wrapped in a transaction (I need to make the same change for the registry's fetch operation).

Reply:
Updated!